### PR TITLE
fix: Set the sslFactory properly for the SR REST client.

### PR DESCRIPTION
**Pull request by jnh5y**
### Description 
_What behavior do you want to change, why, how does your patch achieve the changes?_

### Testing done 
Manual testing is in progress.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-engine/src/main/java/io/confluent/ksql/schema/registry/KsqlSchemaRegistryClientFactory.java [hunk 1]
```diff
@@ -115,17 +115,18 @@ public SchemaRegistryClient get() {
...
-    return schemaRegistryClientFactory.create(
+    final SchemaRegistryClient client = schemaRegistryClientFactory.create(
         restService,
         1000,
         ImmutableList.of(
             new AvroSchemaProvider(), new ProtobufSchemaProvider(), new JsonSchemaProvider()),
         schemaRegistryClientConfigs,
         httpHeaders
     );
+    // Let's reset the sslContext...
```
Comment (suhas-satish) [inline ksqldb-engine/src/main/java/io/confluent/ksql/schema/registry/KsqlSchemaRegistryClientFactory.java:126]: just for my understanding, what happens if this sslContext is not reset?

Comment (jnh5y) [reply ksqldb-engine/src/main/java/io/confluent/ksql/schema/registry/KsqlSchemaRegistryClientFactory.java:126]: The code above `schemaRegistryClientFactory.create(...)` calls https://github.com/confluentinc/schema-registry/blob/master/client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java#L181-L232

The last lines of that method call `restService.setSslSocketFactory(...)`...  effectively, that undoes the code that was setting the factory before we called the constructor.

**Review by spena — APPROVED** [review]
Looks good. Can you just make a comment in the code the reason to reset the SSL factory? In case we look at it in the future.

**Comment by jnh5y** [comment]
> Looks good. Can you just make a comment in the code the reason to reset the SSL factory? In case we look at it in the future.

Yes, will do.

**Pull request merged**